### PR TITLE
Set python version when unset

### LINKED ISSUE #1388: Bootstrap doesn't work for Debian 10 (Buster)
### Description of Issue/Question
Debian 10 Salt packages are only available for Python 3.
But, the Debian 10 default is Python 2.
Bootstrap Python version detection find 2, and try to install packages for this version.
  
### Setup
```
$ docker run --rm -it debian:buster-slim bash
root@b153991e056a:/# apt-get update
root@b153991e056a:/# apt-get -y install curl
root@b153991e056a:/# curl -L https://bootstrap.saltstack.com -o install_salt.sh
```

### Steps to Reproduce Issue
```
root@b153991e056a:/# sh install_salt.sh -P

 *  INFO: Running version: 2019.11.04
 *  INFO: Executed by: 
 *  INFO: Command line: 'install_salt.sh -P'

 *  INFO: System Information:
 *  INFO:   CPU:          GenuineIntel
 *  INFO:   CPU Arch:     x86_64
 *  INFO:   OS Name:      Linux
 *  INFO:   OS Version:   4.9.0-11-amd64
 *  INFO:   Distribution: Debian 10.1

 *  INFO: Installing minion
 *  INFO: Found function install_debian_deps
 *  INFO: Found function config_salt
 *  INFO: Found function preseed_master
 *  INFO: Found function install_debian_stable
 *  INFO: Found function install_debian_restart_daemons
 *  INFO: Found function daemons_running
 *  INFO: Found function install_debian_check_services
 *  INFO: Running install_debian_deps()
[...]
The following NEW packages will be installed:
  ..... python python-minimal python-yaml
  python2 python2-minimal python2.7 python2.7-minimal .....
[...]
The following NEW packages will be installed:
  apt-transport-https .....
[...]
Warning: apt-key output should not be parsed (stdout is not a terminal)
gpg: no valid OpenPGP data found.
 * ERROR: Failed to run install_debian_deps()!!!
```
Python 2.7 and wrong repository:
```
root@b153991e056a:/# cat /etc/apt/sources.list.d/saltstack.list 
deb https://repo.saltstack.com/apt/debian/10/amd64/latest buster main
```
It is not a major issue, but `apt-transport-https` is now useless for Debian 10.
apt include it.

### Idea
Since packages are only available for Python 3. I think the Python version detection is not necessary anymore  for Debian 10 and we could just set version to 3.

And we also can remove `apt-transport-https` from dependencies.

Do you agree with that ?

### PR DESCRIPTION
**Pull request by wolfpackmars2**
Sets python version to 3 if version to install is >= 2019.2 and python3 is available.
fixes #1388 #1418 #1350 
Also sets default to python3 version of Salt
Tested on Debian 10

### What does this PR do?
Sets python3 version of Salt to be installed if not specified on command line and Python3 is available.

### What issues does this PR fix or reference?
#1388 #1418 #1350 and possibly others

### Previous Behavior
Remove this section if not relevant
Couldn't install on Debian 10 without adding -x option
Ubuntu 18.04 would install Python2 Salt by default.

### New Behavior
Defaults to Python3 Salt when certain conditions are met.

### CODE REVIEW

File: bootstrap-salt.sh [hunk 1]
```diff
@@ -785,6 +785,20 @@ if [ -d "${_VIRTUALENV_DIR}" ]; then
     exit 1
 fi
 
+# Set the python exe if unset and version is >= 2019.2
+if [ "$_PY_EXE" = "" ] && [ "$(echo "$STABLE_REV" | grep -E '^(latest|2019\.2|3000)$')" != "" ]; then
+    which python3 > /dev/null
```
Comment (s0undt3ch) [inline bootstrap-salt.sh]: Use `command` instead of `which`. See source code for examples.

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

File: bootstrap-salt.sh (hunk 1 above)
Comment (vutny) [reply bootstrap-salt.sh]: There is a special function for that: `__check_command_exists`

Comment (s0undt3ch) [reply bootstrap-salt.sh]: Even better :smile:

**Comment by bryceml** [comment]
Ideally this would be able to automatically handle any new release of salt being released, i.e. 3002, 3003, etc.  It also becomes much less complicated if we only did this on 3001 and forward as we wouldn't change any working previous behavior.  If we did it on 3001 and newer we also wouldn't have to make special cases for centos6, amazon1 and debian8 only having py2 available.

**Comment by bryceml** [comment]
superceded by https://github.com/saltstack/salt-bootstrap/pull/1577

**Pull request closed without merging**